app/Core/Repository/UpgradeFile.cs

```diff
@@ -219,5 +219,73 @@ public List<KeyValuePair<int, string>> Populate(string query, Dictionary<string,
             }
             finally { this.connection.Close(); }
         }
+
+        public int ExecuteQuery(string query, Dictionary<string, object> parameters)
+        {
+            try
+            {
+                this.Connect();
+                using (MySqlCommand cmd = new MySqlCommand(query, this.connection))
+                {
+                    // Add parameters to the command
+                    foreach (KeyValuePair<string, object> kvp in parameters)
+                    {
+                        cmd.Parameters.AddWithValue(kvp.Key, kvp.Value);
+                    }
+
+                    // Execute the command and return the number of affected rows
+                    return cmd.ExecuteNonQuery();
+                }
+            }
+            catch (MySqlException ex)
+            {
+                // Log or handle specific MySql errors here
+                throw new Exception($"Error executing query: {ex.Message}");
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Unexpected error: {e.Message}");
+            }
+            finally
+            {
+                this.connection.Close(); // Ensure the connection is closed
+            }
+        }
+
+        public int ExecuteScalar(string query, Dictionary<string, object> parameters)
```

Comment:
Redundant code, you can improve the existing method in line 77, if you want to return the scalar method of the query